### ISSUE TITLE
Default handling of zoom levels for new adaptive hillshading

### ISSUE THREAD

**Issue by SammysHP** [mapsforge/mapsforge#1606]
There appears to be a disagreement about the default settings for the new adaptive hillshading implementation. More precisely at which zoom levels it should be enabled by default.

You can read the full discussion here: https://github.com/cgeo/cgeo/issues/16438

Maybe @ztNFny and @Sublimis can summarize their standpoints here to make it easier for others to join the discussion.

**Comment by Sublimis** [comment]
Adaptive hill shading algorithm **_default_** zoom limits:

* 1" files: 0 to 17.
* 3" files: 0 to 15.

This is not a bug. This _feature_ works exactly as the original authors intended, and is easily adjustable.

We are inclined to remove the top limits, **_by default_**, if @devemux86 and others agree.
Also we probably won't participate in further _discussions_ on this, as our standpoint is more than clear, and the source code can speak for itself better than anyone.

**Comment by devemux86** [comment]
> We are inclined to remove the top limits, by default, if @devemux86 and others agree.

@Sublimis Do you mean to just change the default setting?
I don't have a problem with this.

**Comment by Sublimis** [comment]
@devemux86 To make the maximum zoom level not limited at all, by default, in the adaptive algorithm. But let's leave this for another day.

**Comment by ztNFny** [comment]
"Most" old implementations used a limit of 9-17 for "reasons". The reason for the lower limit (9) probably was performance as zooming out requires loading and generating a lot of data. The performance is addresses by the Adaptive implementation and thus the lower limit was removed.
As for the upper limit my best guess is that it was introduced to avoid "ugly" graphics as the resolution of HS data is limited with the specific value of 17 being an arbitrary pick where HS bitmaps are "still good enough". Performance for sure shouldn't be a reason here as the amount of data to be rendered is the same for 17+ as for 17.
Adaptive changes that limit to 14/15 resp 16/17 (3" vs 1", HQ off vs HQ on) - as for the reason for that change, even after the long discussion in cgeo, I still don't understand the why.

In *my opinion and the usecases most known to me*, e.g. planning a hike or skiing, HS data is the most useful at high zoom levels. E.g. when planning a hike you need want to zoom in a lot to be able to see the small trails (zoom level 15-18).

So the requirement to "a HS layer" would be:
* be available on zoom levels where activities such as hiking are done
* don't drain too much performance on low zoom levels
and maybe a secondary requirement of
* don't be ugly

Because of that I find the current *default* settings of Adaptive HS should be tuned to *not be worse* than "most old implementations" in regards to the 2 main requirements. Thus I'd argue for the maxZoom to be set to *at least* the "reference" value of 17 - but to be honest I don't see any reason to limit it in that direction at all!

Of course quality of the bitmap isn't great when zoomed in, but it's already not great at the current max levels - and if someone wants better quality they can use 1" files. However I find a few pixelated areas on the map that allow estimating the altitude changes a lot less irritating than the HS layer just disappearing, which also changes the general appearance of the map (many colors change and you have to re-orient yourself whenever you pass the threshold).
Maybe I'm missing a good reason for hiding the HS layer at high zoom levels though?

Again - this is about the *default* values, not providing an option to change them as that's already there. My view on defaults is: "This is the best practice values that will work for most situations". If there's no such best-practice-value then it should be a mandatory parameter without a default so that the user *has to* select a value.


level 13:
![image](https://github.com/user-attachments/assets/56f15161-7b5b-4f31-8209-0e1bb8dd2d51)

level 15:
![image](https://github.com/user-attachments/assets/cde9e950-512c-491b-9da6-5233be26a669)

level 17: 
![image](https://github.com/user-attachments/assets/d8deb1b8-dd6c-44f1-98d3-90a718420fbc)

level 20:
![image](https://github.com/user-attachments/assets/32d8038e-7469-4655-a8d6-c3fa144ce822)

no HS:
![image](https://github.com/user-attachments/assets/28462fe8-b271-4164-a4a3-3d1294f1e525)

**Comment by devemux86** [comment]
@Sublimis Should we expect more work here?

**Comment by Sublimis** [comment]
Such a simple adjustment, it's not worth all this.

#1612 

However, it took a while because during testing we encountered a completely unrelated issue.

### The unrelated issue

Sometimes, and only on some maps, the hill shading, or rather the entire map, disappears before the (now default) maximum zoom level of 22. The exact reason is unknown, but it is not hill shading. Debugging revealed that the mapsforge rendering threads, the ones that do vector rendering, are doing something (reading from a file) for a very long time, and never actually finish.

Be warned if you come across this. Maybe someone can figure this out?

**Comment by Sublimis** [comment]
Unrelated issue update: Can be reproduced on the sample Android app `MultiLingualMapViever` (desktop does not seem to be affected) by using the "overview" maps, e.g. `world.map` or `europe.map`. Zoom in until the scale shows 1 m. More detailed maps also don't seem to cause the issue.

**Comment by devemux86** [comment]
For the max zoom, let's discuss it in a new thread.

You can also search the repository for old issues:
- https://github.com/mapsforge/mapsforge/issues/572

**Comment by devemux86** [comment]
Implemented by #1612.

**Comment by Sublimis** [comment]
> You can also search the repository for old issues:
> 
>     * [Map not visible at large zoom levels #572](https://github.com/mapsforge/mapsforge/issues/572)

Yes, at first this sounded exactly the same, but at the end it's being said "This does fix the issue". So it appears to not be the same issue 🥲

**Comment by devemux86** [comment]
Apparently it was tested and corrected at the time,
but many years have passed and the library has changed. :slightly_smiling_face: 

In any case, we should review what is the reasonable max zoom level to support.
I don't know if there are or how many people use Mapsforge at such high zoom levels (> 20 or 22).
